core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -50,22 +49,26 @@ import org.apache.spark.util.{AccumulatorV2, Clock, SystemClock, Utils}
  *                        task set will be aborted
  */
 private[spark] class TaskSetManager(
-    sched: TaskSchedulerImpl,
+    val sched: TaskSchedulerImpl,
+    val blacklistTracker: Option[BlacklistTracker],
     val taskSet: TaskSet,
     val maxTaskFailures: Int,
-    clock: Clock = new SystemClock())
+    val clock: Clock)
   extends Schedulable with Logging {
 
-  val conf = sched.sc.conf
+  def this(
+      sched: TaskSchedulerImpl,
+      taskSet: TaskSet,
+      maxTaskFailures: Int,
+      clock: Clock = new SystemClock()) {
+    this(sched, None, taskSet, maxTaskFailures, clock)
```

Comment:
only a really minor reason -- multiple default params don't work as nicely, all the call sites which only specify a `clock` now have to say `clock = clock`.  (while with multiple constructors, the types are enough to figure out what to do.)  but this doesn't really matter, I'll update.